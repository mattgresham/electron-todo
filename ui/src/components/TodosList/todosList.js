import React, { useEffect, useState } from 'react';
import AddTodo from '../AddTodo';
import Todo from '../Todo';
import {
  addTodo,
  completeTodo,
  deleteTodo,
  getTodos
} from '../../services/todo';
import styles from './todosList.module.css';

const TodosList = () => {
  const [todoText, setTodoText] = useState('');
  const [todos, setTodos] = useState([]);
  useEffect(() => {
    getTodos(setTodos);
  }, []);

  return (
    <div className={styles.todoList}>
      <h1 className={styles.heading}>To Do List</h1>
      <hr className={styles.line} />
      <div className={styles.todoContainer}>
        <div>
          {todos.map(todo => (
            <Todo
              {...todo}
              key={todo.id}
              completeTodo={completeTodo(setTodos)}
              deleteTodo={deleteTodo(setTodos)}
            />
          ))}
        </div>
      </div>
      <AddTodo {...{ addTodo, setTodos, setTodoText, todoText }} />
    </div>
  );
};

export default TodosList;
