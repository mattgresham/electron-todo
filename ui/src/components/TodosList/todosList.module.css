.todoList {
  background-color: #FFF;
  border-radius: 0.75rem;
  box-shadow: 0 0 32px -10px rgba(0,0,0,0.7);
  display: flex;
  flex-direction: column;
  height: 100%;
  margin: 0 auto;
  max-height: 40rem;
  max-width: 40rem;
  min-height: 20rem;
  min-width: 20rem;
  padding: 1rem;
  width: 100%;
}

.heading {
  text-align: center;
}

.line {
  border: 1px solid #E3E3DE;
  width: 100%;
}

.todoContainer {
  flex: 1;
  overflow: auto;
  padding: 1rem 0;
}
