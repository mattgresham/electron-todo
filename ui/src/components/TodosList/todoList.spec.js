import React from 'react';
import { render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import TodosList from './';

configure({adapter: new Adapter()});

it(`TodoList renders with default props`, () => {
  const tree = render(<TodosList />);
  expect(toJson(tree)).toMatchSnapshot();
});

