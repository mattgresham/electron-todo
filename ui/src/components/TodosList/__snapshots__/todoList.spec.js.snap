// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders with default props 1`] = `
<div
  class="todoList"
>
  <h1
    class="heading"
  >
    To Do List
  </h1>
  <hr
    class="line"
  />
  <div
    class="todoContainer"
  >
    <div />
  </div>
  <form
    class="container"
  >
    <input
      class="textField"
      placeholder="Enter something to do..."
      type="text"
      value=""
    />
    <button
      class="addButton"
      disabled=""
      type="submit"
    >
      Add
    </button>
  </form>
</div>
`;
